### ISSUE TITLE
`copy.replace` rejects Pydantic models: missing synthesized `__replace__`

### ISSUE THREAD

**Issue by Jeremiah-England** [astral-sh/ty#4095]
### Summary

`copy.replace()` works at runtime on Pydantic `BaseModel` instances (via Pydantic's `__replace__` → `model_copy`), and Pyright accepts it. Ty rejects the call because it does not see a `__replace__` method on the model type.

#### Reproduction

```python
from copy import replace

from pydantic import BaseModel


class M(BaseModel, frozen=True):
    x: int


def f(m: M) -> M:
    return replace(m, x=2)
```

```text
error[invalid-argument-type]: Argument to function `replace` is incorrect
  --> repro.py:11:20
   |
11 |     return replace(m, x=2)
   |                    ^ Expected `_SupportsReplace[Unknown]`, found `M`
   |
info: type `M` is not assignable to protocol `_SupportsReplace[Unknown]`
info: └── protocol member `__replace__` is not defined on type `M`
```

#### Why this happens (Grok 4.5 in Cursor explanation)

typeshed types `copy.replace` as requiring `_SupportsReplace` (a `__replace__` method).

Pydantic implements `__replace__` at runtime, but intentionally hides it from type checkers and expects them to synthesize it from `@dataclass_transform()`:

https://github.com/pydantic/pydantic/blob/v2.13.4/pydantic/main.py#L1140-L1143

```python
# Because we make use of `@dataclass_transform()`, `__replace__` is already synthesized by
# type checkers, so we define the implementation in this `if not TYPE_CHECKING:` block:
def __replace__(self, **changes: Any) -> Self:
    return self.model_copy(update=changes)
```

(`ModelMetaclass` is decorated with `@dataclass_transform(...)`: https://github.com/pydantic/pydantic/blob/v2.13.4/pydantic/_internal/_model_construction.py#L82)

pyright synthesizes `__replace__` for Pydantic models and accepts `copy.replace`. ty's dedicated Pydantic support appears not to, so the model fails the `_SupportsReplace` protocol check.

#### Environment

- ty 0.0.63
- pydantic 2.13.4
- Python 3.14

#### Version

ty 0.0.63

**Comment by carljm** [comment]
This looks like a regression from when we introduced dedicated pydantic support and switched pydantic models from `CodeGeneratorKind::DataclassLike` to `CodeGeneratorKind::Pydantic`, which implicitly then stopped adding `__replace__` for them. Should be an easy fix.

Thanks for the report!
